### PR TITLE
OAK-11206: remove usage of Queues.newArrayDeque

### PR DESCRIPTION
**Pull request by t-rana**

### CODE REVIEW

**Comment by rishabhdaim** [comment]
@t-rana could you please split the PR into 2 parts i.e. one for creating a Utility and the other which uses that.

**Comment by t-rana** [comment]
@rishabhdaim 
I have raised another PR for adding util, once that is merged, I will move this to review again.
https://github.com/apache/jackrabbit-oak/pull/1808

**Comment by rishabhdaim** [comment]
@t-rana you need to remove the utils related changes from this PR.

**Comment by t-rana** [comment]
@rishabhdaim 
I was waiting for my other PR (https://github.com/apache/jackrabbit-oak/pull/1808) to get merged. Now that is merged, and I have updated this one accordingly.

File: oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java [hunk 1]
```diff
@@ -171,7 +171,6 @@ public static <T> Set<T> toSet(@NotNull final Iterator<T> iterator) {
         iterator.forEachRemaining(result::add);
         return result;
     }
-
```
Comment (rishabhdaim) [inline oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java]: please remove this change.

Comment (t-rana) [reply oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java]: Done.

**Review by rishabhdaim — APPROVED** [review]

**Pull request merged**